addons/web/static/src/views/list/list_renderer.js

```diff
@@ -1074,9 +1075,16 @@ export class ListRenderer extends Component {
                     );
                     const nextIsGroup = futureRow.classList.contains("o_group_header");
                     const rowTypeSwitched = cellIsInGroupRow !== nextIsGroup;
+                    const isGroupToGroup = cellIsInGroupRow && nextIsGroup;
+                    if (rowTypeSwitched || isGroupToGroup) {
+                        targetIndex = this.lastKnownIndex || 0;
+                    } else {
+                        this.lastKnownIndex = index;
+                    }
+
                     let defaultIndex = 0;
                     if (cellIsInGroupRow) {
-                        defaultIndex = this.hasSelectors ? 1 : 0;
+                        defaultIndex = this.hasSelectors ? targetIndex : 0;
                     }
```

Comment:
```js
const defaultIndex = cellIsInGroupRow ? targetIndex : 0;
```

Reply:
checking hasSelectors is no longer necessary (it was before for the 0/1 fallback)